openedx/adg/admin_dashboard/admin_settings/views.py

```diff
@@ -1,31 +1,159 @@
+    extract_form_error,
+    send_team_member_invitation_email
 )
 from openedx.adg.admin_dashboard.admin_settings.models import Invitation
 from openedx.adg.admin_dashboard.helpers import is_superuser_or_adg_admin
 from openedx.adg.admin_dashboard.views import EnsureAdminUserAuthenticatedMixin
-from openedx.adg.lms.applications.constants import ADG_ADMIN_GROUP_NAME
+from openedx.adg.lms.applications.constants import ADG_ADMIN_GROUP_NAME, EMAIL
 from openedx.adg.lms.applications.models import BusinessLine
 
 
-class AdminAccountListView(EnsureAdminUserAuthenticatedMixin, ListView):
+class BaseListView(EnsureAdminUserAuthenticatedMixin, ListView):
     """
-    `ListView` for active admin accounts.
+    Base view for `AdminAccountListView` and `PendingInvitationListView` to hold common methods and attributes.
     """
 
     queryset = None
-    template_name = 'adg/admin_dashboard/admin_settings/accounts.html'
-    http_method_names = ('get',)
+    http_method_names = ('get', 'post',)
     paginate_by = ADMIN_SETTINGS_PAGE_SIZE
+
+    def post(self, request, *args, **kwargs):
+        """
+        Post method to parse, validate and send invitations to the invitees
+        """
+        data = request.POST.dict()
+        context = self.handle_invitations(data)
+        return render(request, self.template_name, context)
+
+    def handle_invitations(self, data):
+        """
+        Parses the invitations, validates them and saves them in Invitations model and sends invitation emails to the
+        invitees if validated otherwise sends an error.
+
+        Arguments:
+            data (dict): dictionary containing data related to invitations
+
+        Returns:
+            dict: updated context dictionary
+        """
+        data.pop('csrfmiddlewaretoken')
+        invitations = self.parse_invitations(data)
+        invitation_error = self.validate_and_save_invitations(invitations)
+
+        queryset = self.get_queryset()
+        context = self.get_context_data(object_list=queryset)
+        context['invitation_error'] = invitation_error
+        context['invitations'] = json.dumps(invitations)
+        return context
+
+    def parse_invitations(self, invitations_data):
+        """
+        Iterates over the invitations_data dictionary to create a new 2D dictionary in which each key contains a
+        dictionary that corresponds to a single invitation.
+
+        Arguments:
+            invitations_data (dict): dictionary containing all the data related to invitations
+
+        Returns:
+            dict: a 2D dictionary
+            str: error related to email
+        """
+        invitations = {}
+
+        for key, value in invitations_data.items():
+            field, invitation_id = key.split('-')
+
+            if invitation_id in invitations:
+                invitations[invitation_id].update({field: value})
+            else:
+                invitations[invitation_id] = {field: value}
+
+        return invitations
```

Comment:
We can use collections.defaultdict here.

Code will be:
```suggestion
        invitations = defaultdict(dict)

        for key, value in invitations_data.items():
            field, invitation_id = key.split('-')
            invitations[invitation_id].update({field: value})

        return invitations
```